### PR TITLE
Adds new documentation capabilities

### PR DESCRIPTION
**Pull request by colinmorelli**
This PR adds a few new capabilities that we can use in documentation (and SDK) generation.

## Changes
- Adding a `title` property to schemas to coincide with the property of the same name in OpenAPI
- Extensions for codecs to allow adding arbitrary key/values to the OpenAPI spec (prefixed `x-` in the spec)
- Extensions for operations for the same
- A bugfix to how examples are applied to docs (a `null` was being interpreted as no example)
- Adding an `x-wrapped` property to schemas that are generated when a named schema is referenced and a description is applied (more below)

## X-Wrapped Property
Due to a limitation in OpenAPI 3.0 (OpenAPI 3.1 changes this but is not yet widely supported), we're unable to override a description on a reference property. Consider a schema such as:

```
member:
  type: object
  description: This is a member
  properties:
    name:
      type: string
device:
  type: object
  description: This is a device
  properties:
    member: 
      $ref: '#/components/schemas/member'
```

If we generate documentation for this schema, the `member` property will be documented as "This is a member." However, in many cases, we don't want to inherit documentation from the referenced schema. Instead, we'd like to provide documentation specific to what the `member` property means on a device in particular. Thee OpenAPI 3.0 spec does not allow us to add a `description` to a ReferenceSchema. Instead, we can work around this by using the following schema:

```
device:
  type: object
  description: This is a device
  properties:
    member: 
      oneOf:
        - $ref: '#/components/schemas/member'
      description: The member to which this device belongs
```

In other words, we wrap the reference in a union schema with one option. Now, our schema is no longer a reference schema, so we can apply a description to this. This is a sensible (and the recommended) workaround. However, in doing so, we create a new problem: there's no way to differentiate between a "transparent union with a description" and other types of unions. It would be possible to simply look for a union schema with `oneOf.length === 1`, but that may have other ramifications. Instead, this PR adds `x-wrapped` to the generated schema so that we know we can simply unwind the `oneOf`, access the first child schema to know what's going on.

### CODE REVIEW

**Review by danielschwartz — APPROVED** [review]

**Review by werkshy — APPROVED** [review]
👏 great PR description, thanks

**Comment by werkshy** [comment]
Do we have any test coverage of the OpenAPI feature?

**Comment by colinmorelli** [comment]
@werkshy We don't. I'll setup the harness and add tests for this PR and then we can backfill for other functionality as we make more changes.

**Comment by werkshy** [comment]
> @werkshy We don't. I'll setup the harness and add tests for this PR and then we can backfill for other functionality as we make more changes.

I'm not trying to insist on that right now, more pointing it out as tech debt, but if you do have time to add some initial tests that would be valuable and set us up for the future.
